src/modules/job-manager/submit.c

```diff
@@ -51,14 +50,14 @@ int submit_enqueue_one_job (struct queue *queue, zlist_t *newjobs, json_t *o)
         job_decref (job);
         return -1;
     }
-    if (queue_insert (queue, job, &job->queue_handle) < 0) {
+    if (zhashx_insert (active_jobs, &job->id, job) < 0) {
         job_decref (job);
         // EEXIST is not an error - there is a window for restart_from_kvs()
```

Comment:
perhaps this comment needs to be updated given code changes around it (i.e. zhashx_insert doesn't set EEXIST).